packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -175,11 +176,41 @@ function mountElement(vnode: VElement, parent: ParentNode, anchor: Node | null)
         data: { svg },
     } = vnode;
 
+    if (vnode.isStatic) {
+        if (vnode.elm) {
+            // @todo: what to do about the owner for synthetic shadow and fallbackElmHook
+
+            // @todo: cloned subtree is not associated with the proper shadow. The restrictions and styles will be
+            //        ok.
+            //        maybe modify synthetic shadow and get the owner from ancestors?
+            vnode.elm = vnode.elm.cloneNode(true) as Element;
```

Comment:
Is it necessary to always call `cloneNode(true)`? I think the very first time the VNode is rendered, the `elm` has never been used anywhere else, so it's safe to just use `vnode.elm`. This might actually be more performant, assuming that most static nodes are not used a 2nd time.

Reply:
> Is it necessary to always call cloneNode(true)? 

yes, it may be that it is within a different component instances or within a foreach.

> assuming that most static nodes are not used a 2nd time.

yes it will, the hoisted vnode is shared across component instances, but component instances have different elms, that's why the cloneNode

Reply:
Let me rephrase my question: is it possible to avoid the `cloneNode` the _first_ time we reach this line of code? And then mark somewhere that the `elm` has already been used? E.g.

```js
if (vnode.elmWasUsed) {
  vnode.elm = vnode.elm.cloneNode(true)
} else {
  vnode.elmWasUsed = true
}
```

Reply:
if the element was already used, then we would be doing a `patch`, and we do not do anything in patch (the first condition is `n1 !== n2 -> return`

Reply:
let me clarify, there's no way of knowing to which component instance this `vnode.elm` belongs to, we just know that the rendered element will be identical to the one that is previously set.